ProptechOS-Edge-Interface/examples/java/src/main/java/com/proptechos/service/observations/TelemetryGenerator.java

```diff
@@ -18,33 +20,77 @@
   }
 
   public RecMessage generateTelemetry() {
-    final Sensor randomSensor = getRandomSensor();
+    final Sensor temperatureSensor = getTemperatureSensor();
+    final Sensor humiditySensor = getHumiditySensor();
+    final Sensor co2Sensor = getCO2Sensor();
     //we use temperature from actuation, in case
     //when there was no actuation from ProptechOS we use some randomly generated one
     Object temperatureValue = isNull(actuationValue)
         ? generateRandomTemperature()
         : actuationValue;
-    final RecObservation observation = new RecObservation(ZonedDateTime.now(ZoneOffset.UTC),
-        temperatureValue, randomSensor.getQuantityKind(), randomSensor.getSensorId());
+    // for humidity we use randomly generated values
+    Object humidityValue = generateRandomHumidity();
+    // for CO2 we use randomly generated values
+    Object co2Value = generateRandomCO2();
+    final RecObservation temperetureObservation = new RecObservation(ZonedDateTime.now(ZoneOffset.UTC),
+        temperatureValue, temperatureSensor.getQuantityKind(), temperatureSensor.getSensorId());
+    final RecObservation humidityObservation = new RecObservation(ZonedDateTime.now(ZoneOffset.UTC),
+            humidityValue, humiditySensor.getQuantityKind(), humiditySensor.getSensorId());
+    final RecObservation co2Observation = new RecObservation(ZonedDateTime.now(ZoneOffset.UTC),
+            co2Value, co2Sensor.getQuantityKind(), co2Sensor.getSensorId());
+    List<RecObservation> observations = new ArrayList<>();
+    observations.add(temperetureObservation);
+    observations.add(humidityObservation);
+    observations.add(co2Observation);
     return new RecMessage()
         .setDeviceId(deviceConfig.getDeviceId())
-        .setObservations(Collections.singletonList(observation));
+        .setObservations(observations);
   }
 
   private double generateRandomTemperature() {
     return MIN_TEMPERATURE + rand.nextDouble() * 15;
   }
 
-  private Sensor getRandomSensor() {
-    int randomIndex = rand.nextInt(deviceConfig.getSensors().size());
-    return deviceConfig.getSensors().get(randomIndex);
+  private double generateRandomHumidity() {
+    return MIN_HUMIDITY + rand.nextDouble() * 30;
+  }
+
+  private double generateRandomCO2() {
+    return MIN_CO2 + rand.nextDouble() * 500;
+  }
+
+  private Sensor getTemperatureSensor() {
+    return deviceConfig.getSensors().stream()
+            .filter(s -> s.getQuantityKind().equals(TEMPERATURE_SENSOR_QUANTITY_KIND))
+            .findFirst()
+            .get();
+  }
+
+  private Sensor getHumiditySensor() {
+    return deviceConfig.getSensors().stream()
+            .filter(s -> s.getQuantityKind().equals(HUMIDITY_SENSOR_QUANTITY_KIND))
+            .findFirst()
+            .get();
+  }
+
+  private Sensor getCO2Sensor() {
+    return deviceConfig.getSensors().stream()
+            .filter(s -> s.getQuantityKind().equals(CO2_SENSOR_QUANTITY_KIND))
+            .findFirst()
+            .get();
```

Comment:
lots of code duplication, create only one method, that will receive different Predicates.